### PR TITLE
Functional tests for helloworld, kitchensink, and RichFaces quickstarts

### PR DESCRIPTION
**Pull request by okiss**
spring-controlleradvice and spring-asyncrequestmapping should be fixed now. I added tests for RichFaces quickstarts and greeter-spring

### CODE REVIEW

**Comment by rafabene** [comment]
Thanks okiss, EAP Boms should be using EAP 6.1.1 BOMs, not EAP 6.2.0 BOMs. Can you update it, please ?

**Comment by okiss** [comment]
I removed the property for EAP bom version because it was not used in the poms.

**Comment by sgilda** [comment]
@okiss : Would you mind squashing the commits to make it easier to review?

**Comment by sgilda** [comment]
Thanks @okiss : I'm testing these next.

**Comment by sgilda** [comment]
@okiss : These are very cool tests! I was surprised to see the simulator actually open a browser and see the data entered. Do you think we should add a comment in the README file to indicate this will occur? If you agree, I would be glad to add that to the files.

I ran all the functional tests and most succeeded. I did see failures in 3 of the tests. See below:
Testing results:

*\* BUILD SUCCESS **
- greeter-spring
- helloworld-errai
- helloworld-gwt
- helloworld-html5
- helloworld-rf
- kitchensink-angularjs
- kitchensink-spring-asyncrequestmapping
- kitchensink-spring-basic
- kitchensink-spring-controller-advice
- kitchensink-spring-matrix-variables
- kitchensink-spring-mvc-test

*\* BUILD FAiLURE **
- kitchensink-backbone
  Tests in error: 
  testEmptyRegistration(org.jboss.as.quickstarts.kitchensink.test.KitchensinkBackboneTest): Element is not currently visible and so may not be interacted with
  testRegistrationWithBadNameFormat(org.jboss.as.quickstarts.kitchensink.test.KitchensinkBackboneTest): Element is not currently visible and so may not be interacted with
  testRegistrationWithBadEmailFormat(org.jboss.as.quickstarts.kitchensink.test.KitchensinkBackboneTest): Element is not currently visible and so may not be interacted with
  testRegistrationWithBadPhoneFormat(org.jboss.as.quickstarts.kitchensink.test.KitchensinkBackboneTest): Element is not currently visible and so may not be interacted with
  testRegularRegistration(org.jboss.as.quickstarts.kitchensink.test.KitchensinkBackboneTest): Element is not currently visible and so may not be interacted with
- kitchensink-rf
  Tests in error:
  testRegularRegistration(org.jboss.as.quickstarts.kitchensink.test.KitchensinkRichFacesTest): Cannot locate element using: [id_='memberList'] table tbody[id_='tb']
- richfaces-validation
  Tests in error: 
  org.jboss.as.quickstarts.richfaces_validation.test.MemberValidationTest: Could not invoke deployment method: public static org.jboss.shrinkwrap.api.Archive org.jboss.as.quickstarts.richfaces_validation.test.MemberValidationTest.createTestArchive()

**Comment by okiss** [comment]
@sgilda 
I filed JDF-541, JDF-542 and JDF-544, because they caused the test failures you found. Also I added your description to the README files.

**Comment by sgilda** [comment]
Do we want to check this in if they don't work?

**Comment by sgilda** [comment]
@okiss: Do you have an ETA for the bug fixes above? I would prefer to wait to merge these when I know they work.

**Comment by okiss** [comment]
@sgilda JDF-541 and JDF-542 are already fixed, but the JDF-542 fix will only be available when a new version of jboss-wfk-boms is released.
I don't know when JDF-544 will be fixed but I made a workaround for it. It makes the tests run in a bigger window so a desktop version of the quickstart is loaded and now the tests pass.

**Comment by sgilda** [comment]
These functional tests now work:
- kitchensink-backbone 
- richfaces-validation

The `kitchensink-rf` seems to go through all the tests successfully, but final results show this error: 

```
    Tests in error: 
      testRegularRegistration(org.jboss.as.quickstarts.kitchensink.test.KitchensinkRichFacesTest): Cannot locate element using: [id*='memberList'] table tbody[id*='tb']
```

File: kitchensink-rf/functional-tests/src/test/java/org/jboss/as/quickstarts/kitchensink/test/KitchensinkRichFacesTest.java [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    }
+
+    @Test
+    @InSequence(5)
+    public void testRegularRegistration() {
+        browser.get(contextPath.toString());
+        setInputFields(NAME_FORMAT_OK, EMAIL_FORMAT_OK, PHONE_FORMAT_OK);
+        registerButton.click();
+
+        waitModel().until().element(tableMembers).text().contains(NAME_FORMAT_OK);
```
Comment (sgilda) [inline kitchensink-rf/functional-tests/src/test/java/org/jboss/as/quickstarts/kitchensink/test/KitchensinkRichFacesTest.java:263]: I'm not sure, but I think this is the line the kitchensink-rf test is failing on.

**Comment by okiss** [comment]
@sgilda This is because the members table is not updating automatically because of JDF-542. And fix for JDF-542 will be available with a new version of jboss-wfk-boms.

**Comment by sgilda** [comment]
@okiss : Sorry, Now I understand. It looks like JDF-543 was resolved on Oct. 29. I'll check with @rafabene to get an update.

**Comment by rafabene** [comment]
Don't you men JDF-542 ?

**Comment by sgilda** [comment]
@rafabene : Yes, that was a typo. I do mean JDF-542. :-)

**Comment by sgilda** [comment]
I tested the kitchensink-rf quickstart and it sill fails. It looks like @rafabene resolved JDF-542 in late October, so I'm not clear why this is failing  with:

Tests in error: 
  testRegularRegistration(org.jboss.as.quickstarts.kitchensink.test.KitchensinkRichFacesTest): Cannot locate element using: [id_='memberList'] table tbody[id_='tb']

**Comment by okiss** [comment]
JDF-542 was fixed in October, but jboss-wfk-boms used in quickstarts (2.4.0-build-5) was released a month before that, so it doesn't contain the fix. Kitchensink-rf will work when a new version of jboss-wfk-boms is released and used in quickstarts.

**Comment by rafabene** [comment]
2.4.0-build-6 was released at jboss-developer.github.io/temp-maven-repo/
404 was resolved!

**Comment by sgilda** [comment]
@okiss : The kitchensink-rf functional tests now work! Would you like me to merge it?

**Comment by okiss** [comment]
@sgilda Sure! Thank you

**Comment by sgilda** [comment]
@okiss : Should the commits be squashed before they are merged?

**Comment by okiss** [comment]
@sgilda I squashed the commits.

**Comment by sgilda** [comment]
@okiss : I could have done that on the merge but I just wanted to be sure it was okay with you. But thanks! :-)

**Pull request merged**